server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/CICSVisitor.java

```diff
@@ -386,4 +387,14 @@ private Range buildTokenEndRange(Token token) {
             token.getCharPositionInLine() + token.getStopIndex() - token.getStartIndex() + 1);
     return new Range(p, p);
   }
+
+  private CICSCheckUtilityParameters getCheckParams() {
+    CICSCheckUtilityParameters cicsCheckUtilityParameters = new CICSCheckUtilityParameters();
+    if (context.getPreprocessorsDirectives().get("CICS") == null)
+      cicsCheckUtilityParameters.noLengthEnabled = false;
+    else
+      cicsCheckUtilityParameters.noLengthEnabled =
+          context.getPreprocessorsDirectives().get("CICS").contains("NOLENGTH");
```

Comment:
This really needs to be walked through.
```suggestion
    final List<String> opts = context.getPreprocessorsDirectives().get("CICS");
    if (opts == null) return cicsCheckUtilityParameters;
    for (String opt : opts) {
      switch (opt.toUpperCase()) {
      case "LENGTH":
        cicsCheckUtilityParameters.noLengthEnabled = false;
        break;
      case "NOLENGTH":
        cicsCheckUtilityParameters.noLengthEnabled = true;
        break;
      }
    }
```